sql/firefox_desktop_exact_mau28_v1.sql

```diff
@@ -1,14 +1,57 @@
 CREATE OR REPLACE VIEW
-  firefox_desktop_exact_mau28_v1 AS
+  `moz-fx-data-derived-datasets.telemetry.firefox_desktop_exact_mau28_v1` AS
+WITH
+  base AS (
+  SELECT
+    submission_date,
+    SUM(mau) AS mau,
+    SUM(wau) AS wau,
+    SUM(dau) AS dau,
+    sum(IF(country IN ('US', 'GB', 'FR', 'DE', 'CA'), mau, 0)) AS tier1_mau,
+    sum(IF(country IN ('US', 'GB', 'FR', 'DE', 'CA'), wau, 0)) AS tier1_wau,
+    sum(IF(country IN ('US', 'GB', 'FR', 'DE', 'CA'), dau, 0)) AS tier1_dau,
+    SUM(visited_5_uri_mau) AS visited_5_uri_mau,
+    SUM(visited_5_uri_wau) AS visited_5_uri_wau,
+    SUM(visited_5_uri_dau) AS visited_5_uri_dau
+  FROM
+    `moz-fx-data-derived-datasets.telemetry.firefox_desktop_exact_mau28_by_dimensions_v1`
+  GROUP BY
+    submission_date ),
+  -- We use imputed values for the period after the Armag-add-on deletion event;
+  -- see https://bugzilla.mozilla.org/show_bug.cgi?id=1552558
+  imputed_global AS (
+  SELECT
+    `date` AS submission_date,
+    mau
+  FROM
+    `moz-fx-data-derived-datasets.static.firefox_desktop_imputed_mau28_v1`
+  WHERE
+    datasource = 'desktop_global'),
+    --
+      imputed_tier1 AS (
+  SELECT
+    `date` AS submission_date,
+    mau
+  FROM
+    `moz-fx-data-derived-datasets.static.firefox_desktop_imputed_mau28_v1`
+  WHERE
+    datasource = 'desktop_tier1')
+
 SELECT
-  submission_date,
-  SUM(mau) AS mau,
-  SUM(wau) AS wau,
-  SUM(dau) AS dau,
-  SUM(visited_5_uri_mau) AS visited_5_uri_mau,
-  SUM(visited_5_uri_wau) AS visited_5_uri_wau,
-  SUM(visited_5_uri_dau) AS visited_5_uri_dau
+  base.* REPLACE (
+  COALESCE(imputed_global.mau,
+    base.mau) AS mau,
+  COALESCE(imputed_tier1.mau,
+    base.tier1_mau) AS tier1_mau)
 FROM
-  `moz-fx-data-derived-datasets.telemetry.firefox_desktop_exact_mau28_by_dimensions_v1`
-GROUP BY
-  submission_date
+  base
+LEFT JOIN
```

Comment:
In practice it looks like left join should be fine but given we're imputing "missing" data, and the imputed data is tiny, is there a reason to prefer left join over full join here?

Reply:
I think this should either use full joins, or explain why left joins work and are preferred.

Reply:
Changed to use full joins. There are no days actually missing from the underlying dataset due to the 28-day carryover of clients_last_seen, but you're both right that the left join would introduce problems if a day truly were completely missing.